src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/DeleteCompleteFilesTask.java

```diff
@@ -93,11 +97,24 @@ private boolean tryProcessCompleteEnvelope(CloudBlobContainer container, Envelop
             CloudBlockBlob cloudBlockBlob = container.getBlockBlobReference(envelope.getZipFileName());
 
             if (cloudBlockBlob.exists()) {
-                deleted = cloudBlockBlob.deleteIfExists();
-                if (deleted) {
-                    log.info("File deleted. {}", loggingContext);
-                } else {
-                    log.info("File has not been deleted. {}", loggingContext);
+                Optional<String> leaseId = blobManager.acquireLease(
```

Comment:
do we need to add check metadata to delete ?

Reply:
these files are already in COMPLETED state so all lease expirations (if any) have been handled

Reply:
Meta data can stay there even it is completed. I think you do not need to check metadata.

Reply:
This is what I have meant

Reply:
blobManager.acquireLease( ) method checks metadata?